client.go

```diff
@@ -375,6 +384,25 @@ func (c *ovndb) MonitorTables(jsonContext interface{}) (*libovsdb.TableUpdates,
 	return c.client.Monitor(c.db, jsonContext, requests)
 }
 
+// AddLatestSchemaTables adds new tables that needs to be monitored
```

Comment:
why need AddLatestDBSchemaTables ?
Adding new table should be part of NB/SB table order as we keep adding tables on need basis. 

Client can watch on any tables on need basis

Reply:
@noah8713 the Chassis_Private Tables is not present in all releases of OVN. It is only present in December-2020 release. So, we add that table if OVN SB driver returns that table in its schema. Hence, it needs to be dynamic.

Reply:
Seems then such optimization needs to be for any new table coming up. I think we add table api support for any new table being exposed in OVN. We can add a check in CRUD calls if underlying ovn doesnt have such table exposed. Also client has the flexibility to watch on tables based on their underlying ovn version. So little hesitating as to why go-ovn should worry about if table is missing in actual ovn that its connecting to.

Reply:
@noah8713 we will break existing clients connecting to old server if we blindly update the {SB|NB}TablesOrder. Other way to solve would be to always update the  {SB|NB}TablesOrder array and remove the tables that the remote OVN server doesn't support just before we call MonitorTables().

Reply:
@girishmg : Thanks. I think clients can still pin go-ovn to particular commit to support prior/onwards dec 2020 remote OVN releases. I second for the approach of doing check if new table actually exists before calling monitoring tables. I am ok if that can be incremental PR for client.go to ensure that safety check for any new future table addition/deletion in OVN and add chassisprivate to {SB|NB}TablesOrder.

Reply:
Hi @girishmg @noah8713 thanks for the good discussion. I agree with @noah8713 that AddLatestDBSchemaTables() and LatestSBTablesOrder look weird. In fact, I think not only LatestSBTablesOrder, but NBTablesOrder/SBTablesOrder are not needed. They are used only in MonitorTables() for validating if a table is valid. In my opinion the available tables don't need to be hardcoded. I'd suggest removing NBTablesOrder/SBTablesOrder all together, but it should be done in a separate commit.

Reply:
Sounds good. Thanks @hzhou8 . @Pardhakeswar : Please remove the add/latestdb part and just add chassisprivate as a regular table support following other tables source as a ref. @girishmg  and team please feel free to file an incremental PR to remove NB/SBTables in general as an improvement for dynamic table realization.

Reply:
@hzhou8 as I understand go-ovn does not support all the tables in SB/NB since there are no APIs to do CRUD operations against the unsupported tables. 

For example:  If a client using go-ovn requests for OVN SB's Datapath_binding, then go-ovn will request for that table to be monitored and it will get updates from the SB server, however they can't do much with those updates since there are no APIs.

My understanding was that those tables clearly said what tables are supported for CRUD operation by this library.

Reply:
@girishmg I think the APIs should clearly tell what's supported, and when a user specifically request for monitoring a specific table, they must have clear expectation. It doesn't make much sense for a user to request a table that is not even supported by the API. Or maybe your real concern is the default behavior, i.e. when user doesn't specify which tables to monitor, falling back to monitor-all would incur some more waste?

Reply:
@hzhou8 correct the memory footprint of the clients would explode, for example, with Port_Binding and Datapth_Binding tables from OVN SB DB in a large scale cluster.

By default,  this library should monitor the tables for which there is support for CRUD operations and the end users can filter the table further from that supported list. I think the {SB|NB}TablesOrder enabled it.

Now, with ChassisPrivate table added to SBTablesOrder array and if someone upgrades to latest go-ovn library to get the recently added PortGroup and ACL APIs, then they can't use this library against their old release of OVN since it doesn't support ChassisPrivate table.

The initial PR, added ChassisPrivate table only if the SB Schema had support for it.

Reply:
@girishmg Thanks for clarifying and now I understand your concern. So I think it makes sense to keep the NB/SBTablesOrder arrays for the sake of saving cost when users don't explicitly specify what tables are interested. While for the backward compatibility, it is more straightforward to just keep the tables in NB/SBTablesOrder matching the APIs we support and as you proposed: "remove the tables that the remote OVN server doesn't support just before we call MonitorTables()" sounds good to me.

Reply:
@hzhou8 great! @Pardhakeswar can you please take care of this? @noah8713 FYI